datus/tools/mcp_tools/tool_metadata_extractor.py

```
@@ -0,0 +1,447 @@
+        # Cache for tool lists to avoid repeated calls
+        self.tool_list_cache = TTLCache(maxsize=10, ttl=cache_ttl_seconds)
+
+        logger.info("Initialized Tool Metadata Extractor")
+
+    def _get_default_mcp_server(self) -> str:
+        """Get the first available MCP server name from agent config."""
+        try:
+            from datus.agent.agent import Agent
+            if hasattr(Agent, '_default_agent_config') and Agent._default_agent_config:
+                if hasattr(Agent._default_agent_config, 'mcp_servers') and Agent._default_agent_config.mcp_servers:
+                    return list(Agent._default_agent_config.mcp_servers.keys())[0]
```

Ruff (0.14.5)

Prefer `next(...)` over single element slice

Replace with `next(...)`

(RUF015)